lib/active_admin/resource/belongs_to.rb

```diff
@@ -14,23 +15,26 @@ def initialize(key, namespace)
       # The resource which initiated this relationship
       attr_reader :owner
 
-      # The name of the relation
-      attr_reader :target_name
+      # The name of the relations
+      attr_reader :target_names
+
+      def initialize(owner, target_names, options = {})
+        if target_names.size > 1 && !options[:optional]
+          raise NotSupported, "Belongs to multiple parents must be optional"
+        end
 
-      def initialize(owner, target_name, options = {})
         @owner = owner
-        @target_name = target_name
+        @target_names = target_names
         @options = options
       end
 
-      # Returns the target resource class or raises an exception if it doesn't exist
-      def target
-        resource or raise TargetNotFound.new (@options[:class_name] || @target_name.to_s.camelize), namespace
-      end
-
-      def resource
-        namespace.resources[@options[:class_name]] ||
-          namespace.resources[@target_name.to_s.camelize]
+      # Returns the target resource classes or raises an exception if one of the targets doesn't exist
+      def targets
+        target_names.map do |target_name|
+          namespace.resources[@options[:class_name]] ||
```

Comment:
I'm just getting started with this so bear with me if I don't make much sense, but. The `:class_name` option used to refer to the (previously uniq) target. Since now we can have multiple targets, shouldn't the `:class_name` be configurable _per target_? So maybe something like the following would work better?

```ruby
belongs_to :category, author: { class_name: "User" }, optional: true 
```

Reply:
I have noticed that too. However, I haven't seen anyway to specify `class_name` for each optional resource in the `inhertied_resources` API that we are passing the `@options` to. That's why I leave it the way it is. I can support specifying individual `class_name` for each resource in ActiveAdmin domain if you like.

Reply:
Indirectly related to this problem, I think that `inherited_resources` API has to be redesigned to be more intuitive. For example, this code declares that `task` is nested under `project`:

```ruby
belongs_to :project
belongs_to :task
# The same as
belongs_to :project do
  belongs_to :task
end
```

I propose that we change this behaviour to allow specifying multiple resources in the same level as well as specifying options for each of them.

```ruby
class CommentsController < InheritedResources::Base
  # /other_projects/1/comments
  belongs_to :project, optional: true, route_collection_name: 'other_projects'
  # /tasks/1/comments
  belongs_to :task, optional: true, class_name: 'Todo'

  # Nesting behaviour
  belongs_to :manager do
    # /managers/1/projects/1/comments
    belongs_to :project, optional: true
    # /managers/1/tasks/1/comments
    belongs_to :tasks, optional: true
  end
end
```

Reply:
Yes, that's very confusing to me too! I think your proposal makes much more sense, I believe we can could the behaviour on the next major version.

Regarding the first issue, you're right too... We need to extend the `inherited_resources` API to allow options to be passed _per target_. Would you be willing to work on that?

I stumped with this issue while trying to add a sample multiple optional `belongs_to` to our sample application. I believe it's quite common and people will request it as soon as we release this...

Reply:
Unfortunately, I will stop working on Rails or any Ruby related project soon so I won't be able to work on `inherited_resources` API change. On the other hand, since we want to set multiple `belongs_to` resources in the same level, they should be `optional` by default. Also, I doubt that this change will be pretty complex.

Reply:
Too bad, good luck with your next projects with other tools anyways :)

Just to clarify, you won't be able to work on either the API change you proposed nor in my proposal for this issue (`belongs_to :category, author: { class_name: "User" }, optional: true`), right? If that's the case, no big deal, I had a look at `intherited_resources` and as you point out, it should be fairly simple, you did all the hard work already :purple_heart:

Reply:
I took a look at your proposal and I found that it is not as easy as it look. Let's take your example code for demonstration:

```ruby
belongs_to :category, author: { class_name: "User" }, optional: true
```

The call like this will result in:
+ `targets == [:category]`
+ `options == { author: { class_name: "User" }, optional: true }`

If we want to support this behaviour, we need to know with key in the options is the target, and which keys is the real option. We can whitelist the option keys, but if `inherit_resources` API is changed, someone must know to whitelist the option keys in ActiveAdmin API as well. I find this absurd.

There are other alternatives to your proposal:

The first one looks ugly to me:

```ruby
belongs_to :category, { author: { class_name: "User" } }, optional: true
```

The second one would need to redesign the entire API:

```ruby
belongs_to :category, optional: true
belongs_to :author, class_name: "User", optional: true
```

What do you think?

Reply:
Hi @Looooong. Shit, you're right :anguished:

Should we introduce an `optional_belongs_to` helper that automatically assumes `optional: true`? I believe that would work?

Reply:
Then we must ensure that there is only one call to `belongs_to` or `optional_belongs_to` in the resource. Sounds good?

Reply:
I think that makes sense? Currently, multiple `belongs_to` in a resource doesn't really work in any way, right? Or does it lead to the nesting behavior of the inherited_resources API? Sorry, it's been a while and I don't remember this well.

Reply:
Yes, multiple `belongs_to` doesn't work in ActiveAdmin. We can take this chance to enforce that there is only a single `belongs_to` in resource file. Multiple `belongs_to` will cause the later one override all the `belongs_to` configurations before it. About the nesting behaviour, I'm not so sure about that.

Reply:
Sounds like a good plan @Looooong

Reply:
@deivid-rodriguez I'm sorry to be the bearer of bad news, but I find myself in the position where I cannot allocate more time to work on this PR. Would you mind taking over this PR for me?

Reply:
Of course, thanks for letting me know, and thanks for getting this started!